safe_transaction_service/history/serializers.py

```diff
@@ -256,6 +256,12 @@ def save(self, **kwargs):
                 and (user := request.user)
             ):
                 trusted = user.has_perm("history.create_trusted")
+        try:
```

Comment:
No need to always perform this query. You can check `if sender in self.validated_data["safe_owners"]`, and then if not you can get the delegator from `SafeContractDelegate` as you are doing.

Also you don't need to capture the exception, as if `sender` is neither an `owner` nor a `delegate`, we have a bug in the `validate` function